### PR TITLE
Fixed and clarified help documentation

### PR DESCRIPTION
**Pull request by gold2718**
Fix typos and clarify usage

Help text had typos and unclear (or incorrect) explanations. This PR attempts to improve the situation.

User interface changes?: Yes
Better help text

Fixes: #109 

Testing: 
  test removed:
  unit tests:
  system tests:
  manual testing: Hand proofreading of help text

### CODE REVIEW

File: manic/checkout.py [hunk 1]
```diff
@@ -178,8 +170,9 @@ def commandline_arguments(args=None):
 
     Note: 'externals_only' will only process the external's own
     external description file without trying to manage a repository
-    for the component. This is used for retreiving externals for
-    standalone components like cam and clm. If the source root of the
+    for the component. This is used for retrieving externals for
+    standalone components like cam and ctsm which also serve as
+    sub-components within a larger project. If the source root of the
```
Comment (gold2718) [inline manic/checkout.py:175]: Attempt to clarify usage of this feature.

File: manic/checkout.py [hunk 2]
```diff
@@ -219,21 +212,40 @@ def commandline_arguments(args=None):
...
+    following entry:
+
+    [ libx ]
+    local_path = .
+    protocol = externals_only
+    externals = Externals_LIBX.cfg
+    required = True
+
+    Now, %(prog)s will process Externals.cfg and also process
+    Externals_LIBX.cfg as if it was a sub-external.
```
Comment (gold2718) [inline manic/checkout.py:228]: Attempt to clarify usage of this feature.

File: manic/checkout.py [hunk 3]
```diff
@@ -219,21 +212,40 @@ def commandline_arguments(args=None):
...
+
+Operations performed by manage_externals utilities are explicit and
+data driven. %(prog)s will always attempt to make the working copy
+*exactly* match what is in the externals file when modifying the
+working copy of a repository.
+
+If %(prog)s is not doing what you expected, double check the contents
+of the externals description file or examine the output of
+./manage_externals/%(prog)s --status
+
```
Comment (gold2718) [inline manic/checkout.py:248]: Moved this from the help preamble. It feels more appropriate here. Also correct the language a bit.

**Review by gold2718 — COMMENTED** [review]
Some annotation for the reviewers.

**Review by jedwards4b — APPROVED** [review]

**Pull request merged**